### ISSUE TITLE
SearXNG service: Categories are always alphabetically sorted

### ISSUE THREAD

**Issue by xieve** [NixOS/nixpkgs#292648]
### Describe the bug
When specifying `services.searx.settings.categories_as_tabs`, the categories are always alphabetically sorted.

### Steps To Reproduce
1. Add this to system config: 
```Nix
services.searx = {
  enable = true;
  runInUwsgi = true;
  environmentFile = ./searxng.env;
  settings = {
    use_default_settings = true;
    server.secret_key = "@SEARXNG_SECRET@";
    categories_as_tabs = {
      general = {};
      images = {};
      videos = {};
      news = {};
      map = {};
      it = {};
      science = {};
      files = {};
    };
  };
};
```
2. Next to system config, create file `searxng.env`:
```
SEARXNG_SECRET="myverysecretsecret"
```
3. Rebuild

### Expected behavior
The categories should be in the order they are specified in the config file.

### Screenshots
![image](https://github.com/NixOS/nixpkgs/assets/20223093/f969f7cd-f3d6-4a10-8d36-bc2053cdc208)

### Additional context
Nix always sorts keys, which means these keys get sorted before they are converted to JSON. SearXNG cares about the order they are specified in the JSON, and it does explicitly expect the type of `categories_as_tabs` to be `dict`.

### Notify maintainers
@rnhmjoj @999eagle 

### Metadata
```console
$ nix run nixpkgs#nix-info -- -m
 - system: `"x86_64-linux"`
 - host os: `Linux 6.1.77, NixOS, 24.05 (Uakari), 24.05.20240226.13aff9b`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.18.1`
 - nixpkgs: `/nix/var/nix/profiles/per-user/root/channels/nixos`
```

---

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

**Comment by rnhmjoj** [comment]
> The categories should be in the order they are specified in the config file.

Are you sure this is the intended effect? AFAIK the objects in JSON are unordered, so it shouldn't matter in which order you write the name/value pairs.

**Comment by xieve** [comment]
> > The categories should be in the order they are specified in the config file.
> 
> Are you sure this is the intended effect? AFAIK the objects in JSON are unordered, so it shouldn't matter in which order you write the name/value pairs.

Yes, I am sure. I am not sure why, maybe dicts in YAML are supposed to be ordered? Anyway, they are ordered when they are parsed from YAML into a JS object, and this is the way I had set it up before:

```yaml
categories_as_tabs:
  general:
  images:
  videos:
  news:
  map:
  it:
  science:
  files:
 ```
 
![image](https://github.com/NixOS/nixpkgs/assets/20223093/747af05b-c70e-4a74-9378-bc49ecda3a5d)
